src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -21,22 +39,427 @@ public ErrorListInfoBarController(IHost host)
             }
 
             this.host = host;
+
+            this.bindingInformationProvider = host.GetService<ISolutionBindingInformationProvider>();
+            this.bindingInformationProvider.AssertLocalServiceIsNotNull();
         }
 
         #region IErrorListInfoBarController
         public void Reset()
         {
-            // TBD
+            this.AssertOnUIThread();
+            this.ClearCurrentInfoBar();
         }
 
         public void Refresh()
         {
-            // TBD
+            this.AssertOnUIThread();
+
+            // TODO: part of SVS-72, need to call IProjectSystemFilter.SetTestRegex
+            // and specify the regex that you get from IHost.SonarQubeService.GetProperties
+            // before calling to ISolutionBindingInformationProvider which will internally 
+            // use the regex information to determine which projects are filtered and which are not 
+            // all this needs to be on a background thread!
+
+            // We don't want to slow down solution open, so we delay the processing
+            // until idle. There is a possibility that the user might close and open another solution 
+            // when the delegate will execute, and we should handle those cases
+            if (this.IsActiveSolutionBound)
+            {
+                this.InvokeWhenIdle(this.ProcessSolutionBinding);
+            }
         }
         #endregion
 
         #region Non-public API
+        private void AssertOnUIThread()
+        {
+            Debug.Assert(this.host.UIDispatcher.CheckAccess(), "The controller needs to run on the UI thread");
+        }
+
+        private bool IsActiveSolutionBound
+        {
+            get
+            {
+                return this.bindingInformationProvider.IsSolutionBound();
+            }
+        }
+
+        private void InvokeWhenIdle(Action action)
+        {
+            Debug.Assert(action != null);
+
+            this.host.UIDispatcher.BeginInvoke(
+                  DispatcherPriority.ContextIdle,
```

Comment:
Realistically when will we be 'idle'? What is the definition of 'idle'?
If I open a solution and I start working, running tests, debugging, adding classes, using analyser fixes, typing, is that idle? My worry is we won't run until they go for tea.